### PR TITLE
Extension throws an error if the toolkitClientBuilder is not initialized before initial credential validation

### PR DESCRIPTION
**Pull request by bryceitoc9**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Description

Fixing bug where activating the extension would always dump the user's profile--this should only happen if the user has invalid credentials. This is because `ext.toolkitClientBuilder` was not initialized before running the account check--we couldn't create an STS client.

This current change moves the code to a state where this should be correct--validation is done after  `ext.toolkitClientBuilder` has been loaded, and throws an extension-breaking error if it isn't as a signal to the developer.

## Motivation and Context

It's frustrating for a user to have to reconnect every time they activate the extension.

## Related Issue(s)

Fix for #557

## Testing
Tested locally

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -67,18 +67,33 @@ export async function activate(context: vscode.ExtensionContext) {
...
-             const successfulLogin = await UserCredentialsUtils.addUserDataToContext(currentProfile, awsContext)
-             if (!successfulLogin) {
+            try {
+                const successfulLogin = await UserCredentialsUtils.addUserDataToContext(currentProfile, awsContext)
+                if (!successfulLogin) {
+                    await UserCredentialsUtils.removeUserDataFromContext(awsContext)
+                }
+            } catch (err) {
+                const error = err as Error
+                vscode.window.showErrorMessage(
```
Comment (bryceitoc9) [inline src/extension.ts]: Error looks like this: 
credentialserr

File: src/shared/credentials/userCredentialsUtils.ts [hunk 2]
```diff
@@ -132,26 +132,32 @@ export class UserCredentialsUtils {
...
-                const transformedCredentials: ServiceConfigurationOptions = {
-                    accessKeyId: credentials.accessKeyId,
-                    secretAccessKey: credentials.secretAccessKey
-                }
                 sts = ext.toolkitClientBuilder.createStsClient('us-east-1', transformedCredentials)
+            } catch (err) {
+                if (err instanceof Error) {
+                    const error = err as Error
+                    logger.error(error)
+                    throw error
```
Comment (awschristou) [inline src/shared/credentials/userCredentialsUtils.ts]: Move the throw outside the if block. If err is not instance of Error, the code would proceed to line 154.

File: src/shared/credentials/userCredentialsUtils.ts [hunk 3]
```diff
@@ -132,26 +132,32 @@ export class UserCredentialsUtils {
...
+                }
             }
+        }
 
-            const response = await sts.getCallerIdentity()
+        try {
+            // by this time, sts is either defined or error has been thrown; exclamation mark should be safe
+            const response = await sts!.getCallerIdentity()
```
Comment (awschristou) [inline src/shared/credentials/userCredentialsUtils.ts]: Non-null assertion should not be necessary with above comment.

File: src/shared/credentials/userCredentialsUtils.ts [hunk 4]
```diff
@@ -132,26 +132,32 @@ export class UserCredentialsUtils {
...
+                // We can also use the set region if/when we migrate to a single-region experience:
                 // https://github.com/aws/aws-toolkit-vscode/issues/549
-                const transformedCredentials: ServiceConfigurationOptions = {
-                    accessKeyId: credentials.accessKeyId,
-                    secretAccessKey: credentials.secretAccessKey
-                }
                 sts = ext.toolkitClientBuilder.createStsClient('us-east-1', transformedCredentials)
+            } catch (err) {
+                if (err instanceof Error) {
+                    const error = err as Error
```
Comment (awschristou) [inline src/shared/credentials/userCredentialsUtils.ts]: Cast isn't necessary; it is implied once inside an if-instanceof check

File: src/shared/credentials/userCredentialsUtils.ts (hunk 3 above)
Comment (bryceitoc9) [reply src/shared/credentials/userCredentialsUtils.ts]: The linter doesn't like it without the `!`, unfortunately.

File: src/shared/credentials/userCredentialsUtils.ts (hunk 4 above)
Comment (bryceitoc9) [reply src/shared/credentials/userCredentialsUtils.ts]: Removing the if statement altogether. The linter doesn't like `err.message` if the cast isn't there.

File: src/shared/credentials/userCredentialsUtils.ts (hunk 3 above)
Comment (bryceitoc9) [reply src/shared/credentials/userCredentialsUtils.ts]: Turns out it didn't mind? Must've been some weird state on my side.

**Pull request merged**